File: backend/Models/Tasks.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace backend.Models
{
    public class Tasks
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int TaskId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Priority { get; set; }
        public string Status { get; set; }
        public DateTime DueDate { get; set; }
        public DateTime? CompletedDate { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public string Notes { get; set; }
        public ICollection<UserTasks> UserTasks { get; set; }
        public ICollection<JobUserTasks> JobUserTasks { get; set; }
        public ICollection<CampaignUserTasks> CampaignUserTasks { get; set; }
    }
}

File: frontend/src/app/models/tasks.ts
import {CampaignUserTasks} from './campaign';
import {User} from './user';
import {Notes} from './note';
import {Job} from './job';


export interface UserTaskNotes {
  userTaskNoteId: number;
  userId: string;
  taskId: number;
  noteId: number;
  user: User;
  task: Tasks;
  note: Notes;
}
export interface JobUserTasks {
  jobUserTaskId: number;
  jobId: number;
  userId: string;
  taskId: number;
  user: User;
  task: Tasks;
  job: Job;
}

export interface Tasks {
  taskId: number;
  title: string;
  description: string;
  priority: TaskPriority;
  status: TaskStatus;
  dueDate: Date;
  completionDate?: Date;
  userTaskNotes: UserTaskNotes[];
  campaignUserTasks: CampaignUserTasks[];
  jobUserTasks: JobUserTasks[];
}
export enum TaskPriority {
  Low = 'Low',
  Medium = 'Medium',
  High = 'High',
  Critical = 'Critical'
}
export enum TaskStatus {
  Created = 'Created',
  InProgress = 'InProgress',
  Pending = 'Pending',
  Completed = 'Completed',
  Running = 'Running',
  Cancelled = 'Cancelled',
  Failed = 'Failed',
  Queued = 'Queued',
}


